apps/wallet/src/ui/app/components/network-selector/custom-rpc-input/index.tsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import cl from 'classnames';
+import { Field, Formik, Form } from 'formik';
+import { useState, useCallback } from 'react';
+import * as Yup from 'yup';
+
+import { ENV_TO_API } from '_app/ApiProvider';
+import Button from '_app/shared/button';
+import InputWithAction from '_app/shared/input-with-action';
+import Alert from '_components/alert';
+import { useAppSelector, useAppDispatch } from '_hooks';
+import { setCustomRPC } from '_redux/slices/app';
+
+import st from '../NetworkSelector.module.scss';
+
+const MIN_CHAR = 5;
+
+const validation = Yup.object({
+    rpcInput: Yup.string()
+        .required()
+        .label('Custom RPC URL')
+        .min(MIN_CHAR)
+        .test(
+            'validate-url',
+            'Not a valid URL',
+            (value) =>
+                value?.indexOf('http://') === 0 ||
+                value?.indexOf('https://') === 0
+        ),
+});
+
+export function CustomRPCInput() {
+    const placeholder = ENV_TO_API.customRPC?.fullNode;
+
+    const customRPC = useAppSelector(({ app }) => app.customRPC);
+    const [customRPCURL, setCustomRPCURL] = useState<string>(customRPC || '');
```

Comment:
Do we really need to keep a local state of the custom rpc url? Why not just use the one from the store? `setCustomRPC` action will update it right?

Reply:
we do not save to store until a valid RPC endpoint has been added. Plus, we need it for visual purpose.

Reply:
I am not sure I get your point. `customRPCURL` seems to be used only as an initial value for the form. If the user changes the input it uses the internal state of the form and when the user clicks save we update the url in store and here. So we could change 

```
initialValues={{ rpcInput: customRPCURL }}
```
to 
```
initialValues={{ rpcInput: customRPC || '' }}
```
and remove the local state